Marlin/Marlin_main.cpp

```diff
@@ -4708,7 +4707,18 @@ inline void gcode_M109() {
 
       #endif //TEMP_BED_RESIDENCY_TIME > 0
 
+      // Prevent a wait-forever situation if R is misused i.e. M190 R0
+      if (wants_to_cool) {
+        if (temp < 30) break; // always break at 30°
+        // break if cooling times out
+        if (!next_cool_check_ms || ELAPSED(now, next_cool_check_ms)) {
+          if (fabs(temp - old_temp) < 1) break;
+          next_cool_check_ms = now + 5000;
+        }
+      }
+
```

Comment:
`if ((old_temp - temp) < 1.0) break; ` we are cooling!
`old_temp` must be set to temp somewhere. Courrently it's set to 9999 and stays there.
Make clear the temperatures are float.
5 seconds seems to be a magnitude to small. At the beginning the temperature still could rise. I suggest 60000 - 1° per minute.